### ISSUE TITLE
NPE on transaction commit

### ISSUE THREAD

**Issue by BFergerson** [graknlabs/grakn#6033]
## Description

I was facing an issue where Grakn keeps saying "Invalid Type Read: The type 'identifier_access' does not exist". I confirmed that it does exist in `grakn server schema` but it keeps saying it anyway. While trying to get a minimally reproducible example the issue has turned into an NPE. I hope they are related but maybe they arent.

## Environment

1. OS (where Grakn server runs): Ubuntu 20
2. Grakn version (and platform): Grakn Core 2.0.0-alpha
3. Grakn client: java-client 2.0.0-alpha-4

## Reproducible Steps

```groovy
def graknHost = "localhost"
def graknPort = 1729
def graknKeyspace = "grakn"
def graknClient = new GraknClient(graknHost + ":" + graknPort)
if (graknClient.databases().contains(graknKeyspace)) graknClient.databases().delete(graknKeyspace)
graknClient.databases().create(graknKeyspace)
def schemaSession = graknClient.session(graknKeyspace, Grakn.Session.Type.SCHEMA)

//save schema
def tx = schemaSession.transaction(Grakn.Transaction.Type.WRITE)
tx.query().define(Graql.parseQuery(
            """
define
identifier_access sub relation, relates has_identifier_access, relates is_identifier_access;
"""
) as GraqlDefine)
tx.commit()
tx.close()
schemaSession.close()

def dataSession = graknClient.session(graknKeyspace, Grakn.Session.Type.DATA)
def writeTx = dataSession.transaction(Grakn.Transaction.Type.WRITE)
writeTx.query().insert(Graql.parseQuery("""
match
\$self iid 0x966e80267ffffffffffffffd;
\$other iid 0x966e80267ffffffffffffffc;
insert (is_identifier_access: \$self, has_identifier_access: \$other) isa identifier_access;
"""))
writeTx.commit()
writeTx.close()
dataSession.close()

graknClient.close()
```

## Actual Output

```
2021-01-14 15:05:24,595 [ForkJoinPool-1-worker-41] [ERROR] g.core.server.rpc.TransactionStream - java.lang.NullPointerException
grakn.core.common.exception.GraknException: java.lang.NullPointerException
	at grakn.core.common.exception.GraknException.of(GraknException.java:47)
	at grakn.core.rocks.RocksStorage.exception(RocksStorage.java:186)
	at grakn.core.rocks.RocksStorage$TransactionBounded.exception(RocksStorage.java:240)
	at grakn.core.rocks.RocksStorage$Data.exception(RocksStorage.java:270)
	at grakn.core.graph.GraphManager.exception(GraphManager.java:55)
	at grakn.core.concept.ConceptManager.exception(ConceptManager.java:188)
	at grakn.core.query.QueryManager.match(QueryManager.java:82)
	at grakn.core.query.QueryManager.match(QueryManager.java:74)
	at grakn.core.query.QueryManager.insert(QueryManager.java:155)
	at grakn.core.server.rpc.query.QueryHandler.insert(QueryHandler.java:139)
	at grakn.core.server.rpc.query.QueryHandler.handleRequest(QueryHandler.java:80)
	at grakn.core.server.rpc.TransactionRPC.handleRequest(TransactionRPC.java:96)
	at grakn.core.server.rpc.TransactionStream.handleRequest(TransactionStream.java:130)
	at grakn.core.server.rpc.TransactionStream.onNext(TransactionStream.java:82)
	at grakn.core.server.rpc.TransactionStream.onNext(TransactionStream.java:48)
	at io.grpc.stub.ServerCalls$StreamingServerCallHandler$StreamingServerCallListener.onMessage(ServerCalls.java:255)
	at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.messagesAvailableInternal(ServerCallImpl.java:309)
	at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.messagesAvailable(ServerCallImpl.java:292)
	at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1MessagesAvailable.runInContext(ServerImpl.java:782)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NullPointerException: null
	at grakn.core.logic.tool.TypeResolver$ConstraintMapper.getTypeFromIID(TypeResolver.java:395)
	at grakn.core.logic.tool.TypeResolver$ConstraintMapper.convertIID(TypeResolver.java:388)
	at grakn.core.logic.tool.TypeResolver$ConstraintMapper.lambda$convert$1(TypeResolver.java:311)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at grakn.core.logic.tool.TypeResolver$ConstraintMapper.convert(TypeResolver.java:311)
	at grakn.core.logic.tool.TypeResolver$ConstraintMapper.convert(TypeResolver.java:286)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at java.base/java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1085)
	at java.base/java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1085)
	at grakn.core.logic.tool.TypeResolver$ConstraintMapper.<init>(TypeResolver.java:249)
	at grakn.core.logic.tool.TypeResolver.resolveVariablesExhaustive(TypeResolver.java:73)
	at grakn.core.reasoner.Reasoner.lambda$execute$1(Reasoner.java:69)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1603)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at grakn.core.reasoner.Reasoner.execute(Reasoner.java:70)
	at grakn.core.query.Matcher.execute(Matcher.java:99)
	at grakn.core.query.QueryManager.match(QueryManager.java:80)
	... 20 common frames omitted
```

**Comment by haikalpribadi** [comment]
Which client did you use, @BFergerson ?

**Comment by BFergerson** [comment]
Should have been `java-client 2.0.0-alpha-4`

**Comment by haikalpribadi** [comment]
at which point in your reproducible steps did the bug occur?

**Comment by lriuui0x0** [comment]
@BFergerson This doesn't seem to be reproducible with the latest `2.0.0-alpha-2` grakn core. It now instead throws:

```
[THR04] Invalid Thing Read: The thing with IID '0x966e80267ffffffffffffffd' is not found.
```

**Comment by BFergerson** [comment]
Confirmed fix as of core `2.0.0-alpha-2`.
